oozie/src/main/java/org/apache/falcon/oozie/process/SparkProcessWorkflowBuilder.java

```diff
@@ -0,0 +1,210 @@
+        if (StringUtils.isBlank(sparkMasterURL)) {
+            LOG.error("Spark Master URL can'be empty");
+            return null;
+        }
+        sparkAction.setMaster(sparkMasterURL);
+        sparkAction.setName(sparkJobName);
+
+        addPrepareDeleteOutputPath(sparkAction);
+
+        if (StringUtils.isNotEmpty(sparkOpts)) {
+            sparkAction.setSparkOpts(sparkOpts);
+        }
+
+        if (StringUtils.isNotEmpty(sparkClassName)) {
+            sparkAction.setClazz(sparkClassName);
+        }
+
+        List<String> argList = sparkAction.getArg();
+        List<String> sparkArgs = entity.getSparkAttributes().getArgs();
+        if (sparkArgs != null) {
+            argList.addAll(sparkArgs);
+        }
+
+        checkDataFeedsAsArgument(argList, cluster);
+        sparkAction.setJar(addUri(sparkFilePath, cluster));
+
+
+        setSparkLibFileToWorkflowLib(sparkFilePath, entity);
+        propagateEntityProperties(sparkAction);
+
+        OozieUtils.marshalSparkAction(action, actionJaxbElement);
+        return action;
+    }
+
+    private void setSparkLibFileToWorkflowLib(String sparkFile, Process entity) {
+        if (StringUtils.isEmpty(entity.getWorkflow().getLib())) {
+            entity.getWorkflow().setLib(sparkFile);
+        }
+    }
+
+    private void addPrepareDeleteOutputPath(org.apache.falcon.oozie.spark.ACTION sparkAction) throws FalconException {
+        List<String> deleteOutputPathList = getPrepareDeleteOutputPathList();
+        if (deleteOutputPathList.isEmpty()) {
+            return;
+        }
+
+        org.apache.falcon.oozie.spark.PREPARE prepare = new org.apache.falcon.oozie.spark.PREPARE();
+        List<org.apache.falcon.oozie.spark.DELETE> deleteList = prepare.getDelete();
+
+        for (String deletePath : deleteOutputPathList) {
+            org.apache.falcon.oozie.spark.DELETE delete = new org.apache.falcon.oozie.spark.DELETE();
+            delete.setPath(deletePath);
+            deleteList.add(delete);
+        }
+
+        if (!deleteList.isEmpty()) {
+            sparkAction.setPrepare(prepare);
+        }
+    }
+
+    private void propagateEntityProperties(org.apache.falcon.oozie.spark.ACTION sparkAction) {
+        CONFIGURATION conf = new CONFIGURATION();
+        super.propagateEntityProperties(conf, null);
+
+        List<Property> sparkConf = sparkAction.getConfiguration().getProperty();
+        for (CONFIGURATION.Property prop : conf.getProperty()) {
+            Property sparkProp = new Property();
+            sparkProp.setName(prop.getName());
+            sparkProp.setValue(prop.getValue());
+            sparkConf.add(sparkProp);
+        }
+    }
+
+    private void checkDataFeedsAsArgument(List<String> argList, Cluster cluster) throws FalconException {
+        if (entity.getInputs() == null) {
+            LOG.error("Input Feed Entity is null for process : {}" +entity.getName());
```

Comment:
The purpose of mandating the input/output feeds(which has to be passed as argument to Spark application in process entity) to ensure that spark application must read data and write data from/to Falcon specified Feed entities. Moreover I have checked the pig/hive process entity examples and found that input/output to these examples are based on already defined feed entity only.